doc/dev/design/workflows/storage_devices_and_osds.rst

```diff
@@ -0,0 +1,304 @@
+    - fast response
+    - scalable
+
+**GUI**:
+The inventory should be able to be customized in order to show the desired fields of information for each device. Being customizable also the position of each field(column) and the sort order.
+
+The inventory should be filtered using any of the fields present in the list of devices.
+
+A customized inventory list together with the filter and sort order should be able to be stored for easy utilization. In this way we can provide a set of interesting predefined inventory lists. For example:
+
+    - Devices Available
+    - Devices more used (more average iops) (should be an alert/trigger)
+    - Devices biggers than n Gb
+
+The inventory should also provide a way to do directly operations over physical devices:
+
+    * Identify:  Start/stop to blink the identification light
+    * Create OSD: Create an OSD using this disk device if it is available.
+    * Remove OSD: Delete the OSD using this disk device.
+
+2. Add OSDs
+===========
+
+A. Current workflow
+--------------------
+
+**CLI**:
+We can specify specific devices or use a “drive group” specification to create OSD’s in different hosts. By default, the definition of the OSDs to create is “declarative“ unless you use the unmanaged parameter.
+
+.. prompt:: bash #
+
+  ceph orch daemon add osd <host>:device1,device2 [--unmanaged=true]
+  ceph orch apply osd -i <json_file/yaml_file> [--dry-run] [--unmanaged=true]*
+
+**GUI**:
+Implemented in the dashboard section “cluster.OSDs”.
+There is a button to create the OSDs, that presents a page dialog box to select the physical devices that are going to be part of the OSDs as main, WAL or Db devices.
+It Is very difficult to make a selection ( or to understand how to make the selection). This is even worse if your cluster has the same kind of devices, resulting in the weird thing that is not possible to create an OSD using only one storage device (because you cannot select it)
+The problem here is the UI has been designed to work with “drive groups” and not to work for the user. The “drive group” is an abstract concept that must be used only in the background. Users must not be aware of this concept.
+
+B. Proposed workflow
+--------------------
+
+**CLI and GUI**
+
+The utilization of “declarative” drive groups makes it very difficult to understand how to configure OSD’s and the implications. Also make difficult the implementation because the multiple possibilities and the big amount of different conditions that we can find in a production system makes very complex the right evaluation and use of a declarative description of the storage devices used.
+This results in unexpected situations. For example:
+* A disk cleaned can be reused automatically and without any warning for creating a new OSD.
+* New installed disks are used automatically for OSDs (without any warning)
+* Errors trying to recreate OSD’s in disk removed from the system.
```

Comment:
yes.
For me me `unmanaged=true` is the same as imperative behavior

your second point.... one of the "antinatural"  things we are forced to do using the declarative behaviour

Reply:
Aha! I think we need to clarify terminology here.

When I'm talking about a `declarative` approach, I'm mostly talking about the `declarative` nature of the `DriveGroups`

_Describing_ how OSDs should be deployed with attributes, instead of _explicitly_ listing each device(what is considered `imperative`).

Secondly, we're talking about the _behavior_ of cephadm. Whether the system takes action autonomously(what you seem to understand as `declarative`) or needs to be instructed each time (which you seem to understand as `imperative`).

To avoid confusion lets use different terminology for the second case. Ideas?

Autonomous vs Instructive?

Reply:
I think that the drive group specification is not declarative or imperative.. is just a way to define a device selection based on certain criteria.

The question here is more about behavior, the key point here probably is just to change the cephadm default behavior from "automatic" to "instructive".( and probably avoid any attempt to take automatic decisions until we have more intelligence in the tool)